### PR TITLE
React perf tools

### PR DESCRIPTION
**Pull request by jsantell**
This upgrades React to latest and adds a performance recording button when in development mode to dump out stats.

http://i.imgur.com/TzHPd1G.gif

### CODE REVIEW

File: app/ui/browser/actions/developer.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  Perf = {
+    start() {},
+    stop() {},
+    printWasted() {},
+    getLastMeasurements() {},
+  };
+}
+
+export function perfStart() {
+  console.debug('Start recording React performance'); // eslint-disable
```
Comment (victorporof) [inline app/ui/browser/actions/developer.js]: Use /* eslint no-console: 0 */ at the beginning of the file instead.

File: app/ui/browser/views/developerbar.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const DEVELOPER_BAR_STYLE = Style.registerStyle({
+  position: 'fixed',
+  bottom: 0,
+  right: 0,
+  zIndex: 1,
+  padding: '2px 10px',
+  borderTopLeftRadius: '3px',
+  border: '1px solid #999',
+  backgroundColor: '#eee',
+  flex: 1,
```
Comment (victorporof) [inline app/ui/browser/views/developerbar.jsx]: Why do we need the flex: 1 here?

File: app/ui/browser/views/developerbar.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    if (isCurrentlyRecording) {
+      perfStop();
+    } else {
+      perfStart();
+    }
+  }
+
+  render() {
+    if (!BUILD_CONFIG.development) {
+      return (<div></div>);
```
Comment (victorporof) [inline app/ui/browser/views/developerbar.jsx]: You could also return null here instead.

File: package.json [hunk 4]
```diff
@@ -59,8 +59,8 @@
     "mocha-eslint": "2.0.2",
     "mocha-jsdom": "1.1.0",
     "nyc": "6.4.0",
-    "react-addons-test-utils": "0.14.7",
-    "redux-devtools": "3.1.1",
+    "react-addons-perf": "15.0.1",
+    "react-addons-test-utils": "15.0.1",
     "rimraf": "2.5.2",
```
Comment (victorporof) [inline package.json]: Lots of unrelated changes in this file.

**Comment by victorporof** [comment]
lgtm

File: package.json (hunk 4 above)
Comment (jsantell) [inline package.json]: These include upgrades to React and friends to 0.15 (and many required updates to be in sync)
